src/clib/lib/job_queue/slurm_driver.cpp

```diff
@@ -343,14 +346,14 @@ static std::string make_submit_script(const slurm_driver_type *driver,
  small shell which contains the command to run along with possible slurm
  options, and then this script is submitted with the 'sbatch' command.
 */
-void *slurm_driver_submit_job(void *_driver, const char *cmd, int num_cpu,
-                              const char *run_path, const char *job_name) {
+void *slurm_driver_submit_job(void *_driver, std::string cmd, int num_cpu,
+                              fs::path run_path, std::string job_name) {
     auto driver = static_cast<slurm_driver_type *>(_driver);
 
-    auto submit_script =
-        make_submit_script(driver, cmd, job_name, num_cpu, run_path);
+    auto submit_script = make_submit_script(
+        driver, cmd.c_str(), job_name.c_str(), num_cpu, run_path.c_str());
     std::vector<std::string> sbatch_argv = {
-        "-D" + std::string(run_path), "--job-name=" + std::string(job_name),
+        "-D" + run_path.string(), "--job-name=" + std::string(job_name),
```

Comment:
Did you mean `run_path` instead of `run_path.string()` ? Also job_name is already of type string.

Reply:
Since run_path is a fs::path, `string()` converts it to `std::string`. Not sure if it is technically necessary.

Reply:
Fixed so that job_name is not passed to std::string constructor.